107/src/main/java/org/ehcache/jsr107/internal/DefaultJsr107Service.java

```diff
@@ -19,23 +19,26 @@
 import org.ehcache.core.spi.service.StatisticsService;
 import org.ehcache.jsr107.config.ConfigurationElementState;
 import org.ehcache.jsr107.config.Jsr107Configuration;
-import org.ehcache.jsr107.config.Jsr107Service;
+import org.ehcache.jsr107.Jsr107Service;
 import org.ehcache.spi.service.ServiceDependencies;
 import org.ehcache.spi.service.ServiceProvider;
 import org.ehcache.spi.service.Service;
 
+import static java.util.Objects.requireNonNull;
+
 @ServiceDependencies(StatisticsService.class)
 public class DefaultJsr107Service implements Jsr107Service {
 
   private final Jsr107Configuration configuration;
+  private StatisticsService statisticsService;
```

Comment:
Shouldn't that field be volatile?

Reply:
Probably not since there the ServiceLocator creates a happens before relationship between the start of a dependency and the start of it's dependents... but it's probably safer to add it anyway.